gems/aws-sdk-core/lib/aws-sdk-core/sso_credentials.rb

```diff
@@ -114,12 +114,29 @@ def initialize(options = {})
       end
 
       @async_refresh = true
+      @metrics_source = :code
       super
     end
 
     # @return [SSO::Client]
     attr_reader :client
 
+    def metrics
+      case @metrics_source
+      when :new
```

Comment:
`:new` is not good naming here - what happens when a "version 3" is introduced? Legacy was kind of unfortunate naming too. I'd maybe call this `:session` because it relies on sso_session?